app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx

```diff
@@ -33,24 +32,83 @@ interface Props extends RouteComponentProps<any> {
   isNewsfeedReady: boolean,
   isCoinlistLoading: boolean,
   isCoinlistReady: boolean,
-  fetchNewsItemsForCoin: (coinSlug: string) => void,
-  fetchAllNewsItems: () => void, 
-  fetchMoreNewsItems: () => any, 
+  fetchNewsItemsForCoin: (coinSlug: string) => Promise<NewsItem[]>,
+  fetchAllNewsItems: () => Promise<NewsItem[]>, 
+  fetchMoreNewsItems: () => Promise<NewsItem[]>,
+  fetchNewNewsItems: () => Promise<NewsItem[]>,
 };
 
 interface State {
   initialRenderTips: boolean,
   newsfeedTips: boolean,
+  unseenNews: number[],
+  isWindowFocused: boolean,
 };
 
 class NewsfeedPage extends React.Component<Props, State> {
+
+  private documentTitle = document.title;
+
   state = {
     initialRenderTips: false,
     newsfeedTips: true,
+    unseenNews: [],
+    isWindowFocused: true,
+  }
+
+  startPollingNews = () => {
+    setTimeout(() => {
+      this.fetchNewNewsItems().then(() => {
+        this.startPollingNews();
+      })
+    }, POLLING_TIMEOUT)
+  }
+
+  updateTitle = (news?: NewsItem[]) => {
+    if(typeof news !== 'undefined') {
+      if (news.length === 0) return;
+      document.title = news.length + ' | ' + this.documentTitle;
+      return;
+    }
+
+    if (this.state.unseenNews.length === 0) {
+      document.title = this.documentTitle;
+    } else {
+      document.title = this.state.unseenNews.length + ' | ' + this.documentTitle;
+    }
+  }
+
+  onNewsItemShown = (id) => {
+    setTimeout(() => {
+      this.setState(state => ({ unseenNews: state.unseenNews.filter(_id => _id !== id) }), this.updateTitle);
+    }, 500)
+  }
+
+  fetchNewNewsItems = () => {
+    return this.props.fetchNewNewsItems().then(news => {
+      return new Promise((resolve, _reject) => {
+        if (!this.state.isWindowFocused) {
+          const ids = news.map(elem => elem.id)
+          const unseenNews = _.uniq(this.state.unseenNews.concat(ids))
```

Comment:
i think this const would be more aptly named `unseenNewsIds` because it's an array of ids and not actual news items.

Reply:
Changed to `unseenNewsIds`